tests/run/i4404c.scala

```diff
@@ -0,0 +1,23 @@
+import java.lang.reflect.Modifier
+
+trait T {
+  def f = {
+    final def f0 = ()
```

Comment:
Can you remove the `final` modifier since I believe being able to write it in the first place is a bug?

Reply:
Then, do I need to change the test to use `object` inside a method again?

Reply:
No I think the test is fine as is. We want to make sure that we don't add the `final` modifier in this case